### PR TITLE
Add AutoMapper extension

### LINKED ISSUE #83: Add AutoMapper Extension
In this issue I'm gonna add an extension for injecting automapper with some default profiles for common conversions.

### PR DESCRIPTION
**Pull request by Reza-Noei**
ITNOA

Please review this PR 🌹

fixes #83

### CODE REVIEW

**Comment by soroshsabz** [comment]
Please add README.md to explain how to use it in client projects

File: Source/BSN.Commons.AutoMapper/BSN.Commons.AutoMapper.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
```
Comment (soroshsabz) [inline Source/BSN.Commons.AutoMapper/BSN.Commons.AutoMapper.csproj:31]: Please add packaing information (like other commons project)

File: Source/BSN.Commons.AutoMapper/BSN.Commons.AutoMapper.csproj [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (soroshsabz) [inline Source/BSN.Commons.AutoMapper/BSN.Commons.AutoMapper.csproj:1]: Please update cake for building and packaing this project correctly

File: Source/BSN.Commons.AutoMapper/Extensions/IServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+﻿using AutoMapper;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace BSN.Commons.AutoMapper.Extensions
```
Comment (soroshsabz) [inline Source/BSN.Commons.AutoMapper/Extensions/IServiceCollectionExtensions.cs:4]: Please add some test project to ensure this behavior work correctly with downstream AutoMapper profiles.

**Review by soroshsabz — CHANGES_REQUESTED** [review]
Please see comments and resolve it

**Review by soroshsabz — APPROVED** [review]

File: Source/BSN.Commons.AutoMapper/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# BSN.Commons.AutoMapper
+
+ITNOA
+
+This package contains some facilities for using AutoMapper in Enterprise Way
+
+AutoMapper is a popular library for mapping objects from one type to another. It simplifies the process of mapping complex objects and reduces the amount of code needed to perform these mappings.
```
Comment (soroshsabz) [inline Source/BSN.Commons.AutoMapper/README.md]: Please add explicitly description of supported mapping complex types currently.

File: Test/BSN.Commons.AutoMapper.Tests/CommonMapperProfileTests .cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
+﻿using AutoMapper;
+using BSN.Commons.AutoMapper;
+using BSN.Commons.Responses;
+
+namespace BSN.Commons.Tests
+{
+    [TestFixture]
+    public class CommonMapperProfileTests
```
Comment (soroshsabz) [inline Test/BSN.Commons.AutoMapper.Tests/CommonMapperProfileTests .cs]: Please add one another test, that shows working this AutoMapper helper with additional custom AutoMapper profile

**Review by soroshsabz — CHANGES_REQUESTED** [review]

**Comment by soroshsabz** [comment]
@MostafaFarajbakhsh

**Review by Reza-Noei — COMMENTED** [review]
Code reviewed.

**Review by Reza-Noei — COMMENTED** [review]
Code successfully reviewed.

**Review by soroshsabz — APPROVED** [review]

**Pull request merged**